### PR TITLE
Simplify build with tshy

### PR DESCRIPTION
**Pull request by chrmod**
This should surpass #4077 and #4061 

`tshy` recognises the challanges of publishing hybrid packages and simplify this process.

Requires https://github.com/remusao/mono/pull/835

### CODE REVIEW

**Comment by chrmod** [comment]
One problem remaining with `adblocker-electron-example`:
```
adblocker/packages/adblocker-electron-example on  tshy [$!?] via  v20.10.0
❯ yarn start
[4341:0711/165805.698613:ERROR:CONSOLE(2)] "Unable to load preload script: /Users/chrmod/Projects/github.com/ghostery/adblocker/packages/adblocker-electron-preload/dist/commonjs/index.js", source: node:electron/js2c/sandbox_bundle (2)
[4341:0711/165805.698632:ERROR:CONSOLE(2)] "Error: module not found: @cliqz/adblocker-content", source: node:electron/js2c/sandbox_bundle (2)
```

File: packages/adblocker-electron/src/preload_path-cjs.cts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { resolve } from 'node:path';
```
Comment (chrmod) [inline packages/adblocker-electron/src/preload_path-cjs.cts:1]: This file was meant to be used by `tshy` but is currently not in use. Its safe took keep as a reference.

Comment (chrmod) [reply packages/adblocker-electron/src/preload_path-cjs.cts:1]: https://github.com/isaacs/tshy?tab=readme-ov-file#commonjs-dialect-polyfills

File: packages/adblocker/test/matching.test.ts [hunk 2]
```diff
@@ -11,19 +11,19 @@ import 'mocha';
 
 import { getDomain } from 'tldts-experimental';
 
-import CosmeticFilter from '../src/filters/cosmetic';
-import NetworkFilter, { isAnchoredByHostname } from '../src/filters/network';
+import CosmeticFilter from '../src/filters/cosmetic.js';
```
Comment (remusao) [inline packages/adblocker/test/matching.test.ts:14]: I don't understand why all of these have `.js` extensions although we are importing TypeScript files. That's confusing to me.

Comment (chrmod) [reply packages/adblocker/test/matching.test.ts:14]: From what I understand this is exactly as modern typescript expects it to be when running in `nodenext` mode https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext

**Review by remusao — APPROVED** [review]

**Pull request merged**